app/src/components/integrations/modals/addLdapIntegrationModal/addLdapIntegrationModal.jsx

```diff
@@ -0,0 +1,143 @@
+import { defineMessages, useIntl } from 'react-intl';
+import { Modal } from '@reportportal/ui-kit';
+import { withModal } from 'components/main/modal';
+import { hideModalAction } from 'controllers/modal';
+import { FULL_NAME } from 'components/integrations/integrationProviders/ldapIntegration/constants';
+import { INTEGRATION_FORM } from 'components/integrations/elements';
+import { PLUGINS_PAGE_EVENTS } from 'components/main/analytics/events';
+import { TwoStepsContent } from './twoSteps/twoStepsContent';
+import { createTwoStepsFooter } from './twoSteps/twoStepsFooter';
+
+const messages = defineMessages({
+  createLdapTitle: {
+    id: 'AddLdapIntegrationModal.createLdapTitle',
+    defaultMessage: 'Create Global Integration',
+  },
+  editLdapIntegrationTitle: {
+    id: 'AddLdapIntegrationModal.editLdapIntegrationTitle',
+    defaultMessage: 'Edit Global Integration',
+  },
+  serverSettingsTitle: {
+    id: 'AddLdapIntegrationModal.serverSettingsTitle',
+    defaultMessage: 'Server settings',
+  },
+  fieldSettingsTitle: {
+    id: 'AddLdapIntegrationModal.fieldSettingsTitle',
+    defaultMessage: 'Field settings',
+  },
+});
+
+const AddLdapIntegrationModal = ({ data, dirty, valid }) => {
+  const [metaData, setMetaData] = useState({});
+  const [stepNumber, setStepNumber] = useState(1);
+  const [forceConfirm, setForceConfirm] = useState(false);
+  const [closeModal, setCloseModal] = useState(false);
+  const dispatch = useDispatch();
+  const { formatMessage } = useIntl();
+  const { trackEvent } = useTracking();
+  const { onConfirm, customProps } = data;
+
+  const updateMetaData = (newMetaData) => {
+    setMetaData({ ...metaData, ...newMetaData });
+  };
+
+  const isEdit = customProps.editAuthMode;
+
+  const onSubmit = (newData) => {
+    if (stepNumber === 1 && !forceConfirm) {
+      setStepNumber(2);
+    } else if (isEdit) {
+      trackEvent(PLUGINS_PAGE_EVENTS.clickEditLdapIntegration(newData.nameType === FULL_NAME));
+      onConfirm(newData, () => setCloseModal(true), metaData);
+    } else {
+      trackEvent(PLUGINS_PAGE_EVENTS.clickCreateLdapIntegration(newData.nameType === FULL_NAME));
+      onConfirm(newData, metaData);
+    }
+  };
+
+  const handleSubmit = () => dispatch(submit(INTEGRATION_FORM));
+
+  const createTitle = messages.createLdapTitle;
+  const editTitle = messages.editLdapIntegrationTitle;
+
+  const stepTitles = [
+    formatMessage(messages.serverSettingsTitle),
+    formatMessage(messages.fieldSettingsTitle),
+  ];
+
+  const onStepChange = (step) => {
+    if (step > stepNumber) {
+      valid && setStepNumber(step);
+      handleSubmit();
+    } else {
+      setStepNumber(step);
+    }
+  };
+
+  const onDiscard = () => dispatch(reset(INTEGRATION_FORM));
+  return (
+    <Modal
+      title={formatMessage(isEdit ? editTitle : createTitle)}
+      onClose={() => dispatch(hideModalAction())}
+      allowCloseOutside={!dirty}
+      CustomFooter={createTwoStepsFooter(
```

Comment:
I suggest to update the `Modal` component to receive custom footer as a render prop `createFooter`.
In this case you'll directly pass props to your component here like
`createFooter={(closeHandler) => (<Footer {...yourProps} />)}`